ftbfs with GCC-15

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1096795

I don't know the full set of issues, but at the very least,
`src/Elf.h` is missing an `#include <cstdint>`,
and `SHT_LOUSER` enumerator should be `0x80000000U` not `0x80000000`.
 
```
In file included from /build/reproducible-path/halide-19.0.0/src/Elf.cpp:1:
/build/reproducible-path/halide-19.0.0/src/Elf.h:51:20: error: unknown type name 'uint8_t'
   51 |     enum Binding : uint8_t {
      |                    ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:59:17: error: unknown type name 'uint8_t'
   59 |     enum Type : uint8_t {
      |                 ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:69:23: error: unknown type name 'uint8_t'
   69 |     enum Visibility : uint8_t {
      |                       ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:79:5: error: unknown type name 'uint64_t'
   79 |     uint64_t offset = 0;
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:80:5: error: unknown type name 'uint32_t'
   80 |     uint32_t size = 0;
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:115:44: error: unknown type name 'uint64_t'
  115 |     Symbol &define(const Section *section, uint64_t offset, uint32_t size) {
      |                                            ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:115:61: error: unknown type name 'uint32_t'
  115 |     Symbol &define(const Section *section, uint64_t offset, uint32_t size) {
      |                                                             ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:127:5: error: unknown type name 'uint64_t'
  127 |     uint64_t get_offset() const {
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:130:5: error: unknown type name 'uint32_t'
  130 |     uint32_t get_size() const {
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:158:5: error: unknown type name 'uint32_t'
  158 |     uint32_t type = 0;
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:159:5: error: unknown type name 'uint64_t'
  159 |     uint64_t offset = 0;
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:165:16: error: unknown type name 'uint32_t'
  165 |     Relocation(uint32_t type, uint64_t offset, int64_t addend, const Symbol *symbol)
      |                ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:165:31: error: unknown type name 'uint64_t'
  165 |     Relocation(uint32_t type, uint64_t offset, int64_t addend, const Symbol *symbol)
      |                               ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:172:26: error: unknown type name 'uint32_t'
  172 |     Relocation &set_type(uint32_t type) {
      |                          ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:176:5: error: unknown type name 'uint32_t'
  176 |     uint32_t get_type() const {
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:184:28: error: unknown type name 'uint64_t'
  184 |     Relocation &set_offset(uint64_t offset) {
      |                            ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:188:5: error: unknown type name 'uint64_t'
  188 |     uint64_t get_offset() const {
      |     ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:215:17: error: unknown type name 'uint32_t'
  215 |     enum Type : uint32_t {
      |                 ^
/build/reproducible-path/halide-19.0.0/src/Elf.h:230:22: error: enumerator value evaluates to 2147483648, which cannot be narrowed to type 'int' [-Wc++11-narrowing]
  230 |         SHT_LOUSER = 0x80000000,
      |                      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.

```

--- a/src/Elf.h
+++ b/src/Elf.h
@@ -2,6 +2,7 @@
 #define HALIDE_ELF_H
 
 #include <algorithm>
+#include <cstdint>
 #include <iterator>
 #include <list>
 #include <memory>
@@ -225,17 +226,17 @@ public:
         SHT_REL = 9,
         SHT_SHLIB = 10,
         SHT_DYNSYM = 11,
-        SHT_LOPROC = 0x70000000,
-        SHT_HIPROC = 0x7fffffff,
-        SHT_LOUSER = 0x80000000,
-        SHT_HIUSER = 0xffffffff,
+        SHT_LOPROC = 0x70000000u,
+        SHT_HIPROC = 0x7fffffffu,
+        SHT_LOUSER = 0x80000000u,
+        SHT_HIUSER = 0xffffffffu,
     };
 
     enum Flag : uint32_t {
         SHF_WRITE = 0x1,
         SHF_ALLOC = 0x2,
         SHF_EXECINSTR = 0x4,
-        SHF_MASKPROC = 0xf0000000,
+        SHF_MASKPROC = 0xf0000000u,
     };
 
     typedef std::vector<Relocation> RelocationList;
@@ -463,8 +464,8 @@ public:
         ET_EXEC = 2,
         ET_DYN = 3,
         ET_CORE = 4,
-        ET_LOPROC = 0xff00,
-        ET_HIPROC = 0xffff,
+        ET_LOPROC = 0xff00u,
+        ET_HIPROC = 0xffffu,
     };
 
     // We use lists for sections and symbols to avoid iterator

